sources/engine/Stride.Physics/Shapes/StaticMeshColliderShape.cs

```diff
@@ -13,91 +12,147 @@
+        struct DrawEdge
+        {
+            public Vector3 a;
+            public Vector3 b;
 
-        public IReadOnlyList<Vector3> Vertices => vertices;
-        public IReadOnlyList<int> Indices => indices;
+            public static bool operator <(DrawEdge p_edge_a, DrawEdge p_edge_b)
+            {
+                if (p_edge_a.a == p_edge_b.a)
+                {
+                    return p_edge_a.b < p_edge_b.b;
+                }
+                else
+                {
+                    return p_edge_a.a < p_edge_b.a;
+                }
+            }
 
+            public static bool operator >(DrawEdge p_edge_a, DrawEdge p_edge_b)
+            {
+                if (p_edge_a.a == p_edge_b.a)
+                {
+                    return p_edge_a.b > p_edge_b.b;
+                }
+                else
+                {
+                    return p_edge_a.a > p_edge_b.a;
+                }
+            }
 
-        /// <summary>
-        /// Create a static collider from the vertices provided, ICollection will be duplicated before usage, 
-        /// changes to the collection provided won't be reflected on the collider or <see cref="Vertices"/> and <see cref="Indices"/>.
-        /// </summary>
-        public StaticMeshColliderShape(ICollection<Vector3> vertices, ICollection<int> indices, Vector3 scaling) : this(vertices.ToArray(), indices.ToArray(), scaling)
-        {
+            public DrawEdge(Vector3 p_a, Vector3 p_b)
+            {
+                a = p_a;
+                b = p_b;
 
+                if (a < b)
+                {
+                    b = p_a;
+                    a = p_b;
+                }
+            }
         }
 
-        /// <summary>
-        /// Internal constructor, expects readonly array; any changes made to the vertices won't be reflected on the physics shape
-        /// </summary>
-        StaticMeshColliderShape(Vector3[] verticesParam, int[] indicesParam, Vector3 scaling)
+        private readonly IReadOnlyList<Vector3> faces;
+
+        public StaticMeshColliderShape(IReadOnlyList<Vector3> _faces, Vector3 scaling)
         {
             Type = ColliderShapeTypes.StaticMesh;
             Is2D = false;
 
-            vertices = verticesParam;
-            indices = indicesParam;
-            
-            var meshData = new BulletSharp.TriangleIndexVertexArray(indices, new StrideToBulletWrapper(vertices));
-            var baseCollider = new BulletSharp.BvhTriangleMeshShape(meshData, true);
-            InternalShape = new BulletSharp.ScaledBvhTriangleMeshShape(baseCollider, scaling);
+            cachedScaling = scaling;
+
+            faces = _faces;
+
+            var trimesh = new BulletSharp.TriangleMesh();
+            for (var i = 0; i < faces.Count; i += 3)
+            {
+                var a = faces[i];
+                var b = faces[i+1];
+                var c = faces[i+2];
+
+                trimesh.AddTriangle(a, b, c);
+            }
+
```

Comment:
You might want to swap to a vertex+index buffer instead of face buffer to save on memory and improve instantiation of the shape.
[AddTriangle](https://github.com/stride3d/BulletSharpPInvoke/blob/425a3bb927eff0b8585f718a7860882982b02708/BulletSharp/Collision/TriangleMesh.cs#L32) has to go through bullet ``c#`` api, then through the ``c`` side and finally to ``c++`` on each call while the other goes through [a simple loop and copy](https://github.com/stride3d/BulletSharpPInvoke/blob/425a3bb927eff0b8585f718a7860882982b02708/BulletSharp/Collision/TriangleIndexVertexArray.cs#L94), [bullet actually recommends avoiding AddTriangle](https://github.com/bulletphysics/bullet3/blob/f84b523c0583edd5a667ca867d5e533521a3cbbc/src/BulletCollision/CollisionShapes/btTriangleMesh.h#L51).

As for memory consumption, for a lone triangle, ``face`` takes 36 bytes while ``vert+index`` takes 48, face wins in this case but as soon as you start adding neighboring triangles it becomes significantly skewed towards vert+index; a triangle with three others on each side needs ``4 faces * 3 vertices per face * 3 floats per vert * 4 bytes per float = 144 bytes`` of data with face buffers
while ``vert+index`` is ``6 vertices * 3 floats per vert * 4 bytes per float + (4 triangles * 3 ints per tri * 4 bytes per int) = 120 bytes``.

22 bytes less than ``face`` and that's just with three edges shared between them, the larger the ratio of shared edge is the larger the difference in allocated bytes is, and the vast majority of models these days have a far larger amount of triangles sharing some edge than none at all.